libraries/3rdparty/mbedtls_utils/mbedtls_error.c

```diff
@@ -0,0 +1,1583 @@
+
+            if( use_ret == -( MBEDTLS_ERR_X509_INVALID_SERIAL ) )
+            {
+                rc = "X509 - The serial tag or value is invalid";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_INVALID_ALG ) )
+            {
+                rc = "X509 - The algorithm tag or value is invalid";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_INVALID_NAME ) )
+            {
+                rc = "X509 - The name tag or value is invalid";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_INVALID_DATE ) )
+            {
+                rc = "X509 - The date tag or value is invalid";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_INVALID_SIGNATURE ) )
+            {
+                rc = "X509 - The signature tag or value invalid";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_INVALID_EXTENSIONS ) )
+            {
+                rc = "X509 - The extension tag or value is invalid";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_UNKNOWN_VERSION ) )
+            {
+                rc = "X509 - CRT/CRL/CSR has an unsupported version number";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_UNKNOWN_SIG_ALG ) )
+            {
+                rc = "X509 - Signature algorithm (oid) is unsupported";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_SIG_MISMATCH ) )
+            {
+                rc = "X509 - Signature algorithms do not match. (see \\c ::mbedtls_x509_crt sig_oid)";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_CERT_VERIFY_FAILED ) )
+            {
+                rc = "X509 - Certificate verification failed, e.g. CRL, CA or signature check failed";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_CERT_UNKNOWN_FORMAT ) )
+            {
+                rc = "X509 - Format not recognized as DER or PEM";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_BAD_INPUT_DATA ) )
+            {
+                rc = "X509 - Input invalid";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_ALLOC_FAILED ) )
+            {
+                rc = "X509 - Allocation of memory failed";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_FILE_IO_ERROR ) )
+            {
+                rc = "X509 - Read/write of file failed";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_BUFFER_TOO_SMALL ) )
+            {
+                rc = "X509 - Destination buffer is too small";
+            }
+
+            if( use_ret == -( MBEDTLS_ERR_X509_FATAL_ERROR ) )
+            {
+                rc = "X509 - A fatal error occured, eg the chain is too long or the vrfy callback failed";
+            }
+        #endif /* MBEDTLS_X509_USE_C || MBEDTLS_X509_CREATE_C */
+        /* END generated code */
+    }
+
+    return rc;
+}
+
+const char * mbedtls_strerror_lowlevel( int errnum )
+{
+    const char * rc = NULL;
+    int use_ret;
```

Comment:
Same comment about initialization

Reply:
Same as [this reply](https://github.com/aws/amazon-freertos/pull/1875#discussion_r405869555)